lib/src/main/java/space/taran/arklib/domain/index/PlainResourcesIndex.kt

```diff
@@ -43,13 +44,15 @@ internal data class Difference(
 class PlainResourcesIndex internal constructor(
     private val root: Path,
     private val dao: ResourceDao,
-    private val previewStorage: PreviewStorage,
     private val metadataStorage: MetadataStorage,
```

Comment:
We should also decouple index from metadata storage, but it can be done in separate PR.

Reply:
`Index` uses `ResourceMeta`, so we need it immediately after indexing
If `Index` would use only `ResourceId`, then we could remove `MetadataStorage`